Packet++/header/NdpLayer.h

```diff
@@ -17,353 +17,393 @@
-	 * (NdpOption#isNull() == true) if addition failed. In case of a failure a corresponding error message will be
-	 * printed to log
-	 */
-	NdpOption addNdpOption(const NdpOptionBuilder &optionBuilder);
+		NDP_OPTION_SOURCE_LINK_LAYER = 1,
+		NDP_OPTION_TARGET_LINK_LAYER = 2,
+		NDP_OPTION_PREFIX_INFORMATION = 3,
+		NDP_OPTION_REDIRECTED_HEADER = 4,
+		NDP_OPTION_MTU = 5,
+		NDP_OPTION_UNKNOWN = 255
+	};
 
 	/**
-	 * Remove all NDP options from the layer
-	 * @return True if options removed successfully or false if some error occurred (an appropriate error message will
-	 * be printed to log)
+	 * @class NdpOption
+	 * A wrapper class for NDP options. This class does not create or modify NDP option records, but rather
+	 * serves as a wrapper and provides useful methods for retrieving data from them
 	 */
-	bool removeAllNdpOptions();
-
-protected:
-	NDPLayerBase() = default;
-
-	NDPLayerBase(uint8_t *data, size_t dataLen, Layer *prevLayer, Packet *packet)
-		: IcmpV6Layer(data, dataLen, prevLayer, packet)	{}
-
-private:
-	TLVRecordReader<NdpOption> m_OptionReader;
-
-	virtual size_t getNdpHeaderLen() const = 0;
-	virtual uint8_t *getNdpOptionsBasePtr() const { return m_Data + getNdpHeaderLen(); };
-	NdpOption addNdpOptionAt(const NdpOptionBuilder &optionBuilder, int offset);
-};
-
-/**
- * @class NDPNeighborSolicitationLayer
- * Represents a NDP Neighbor Solicitation protocol layer
- */
-class NDPNeighborSolicitationLayer : public NDPLayerBase
-{
-public:
-	/**
-	 * @struct ndpneighborsolicitationhdr
-	 * Represents neighbor solicitation message format
-	 */
-#pragma pack(push, 1)
-	struct ndpneighborsolicitationhdr : icmpv6hdr
+	class NdpOption : public TLVRecord<uint8_t, uint8_t>
 	{
-		/** Reserved */
-		uint32_t reserved;
-		/** Target address - Target address of solicitation message */
-		uint8_t targetIP[16];
+	public:
+		/**
+		 * A c'tor for this class that gets a pointer to the option raw data (byte array)
+		 * @param[in] optionRawData A pointer to the NDP option raw data
+		 */
+		explicit NdpOption(uint8_t* optionRawData) : TLVRecord(optionRawData)
+		{}
+
+		/**
+		 * A d'tor for this class, currently does nothing
+		 */
+		~NdpOption()
+		{}
+
+		/**
+		 * @return NDP option type casted as pcpp::NDPNeighborOptionTypes enum. If the data is null a value
+		 * of NDP_OPTION_UNKNOWN is returned
+		 */
+		NDPNeighborOptionTypes getNdpOptionType() const
+		{
+			if (m_Data == nullptr)
+				return NDPNeighborOptionTypes::NDP_OPTION_UNKNOWN;
+
+			return static_cast<NDPNeighborOptionTypes>(m_Data->recordType);
+		}
+
+		// implement abstract methods
+
+		size_t getTotalSize() const
+		{
+			if (m_Data == nullptr)
+				return (size_t)0;
+
+			return (size_t)m_Data->recordLen * 8;
+		}
+
+		size_t getDataSize() const
+		{
+			if (m_Data == nullptr)
+				return 0;
+
+			return (size_t)m_Data->recordLen * 8 -
+			       (2 * sizeof(uint8_t));  // length value is stored in units of 8 octets
```

Comment:
nit: if we move the comment up the line will probably not break up

Reply:
yes, fixed.